examples/ecs/fallible_systems.rs

```diff
@@ -6,13 +6,53 @@ use bevy::prelude::*;
 use rand::distributions::Distribution;
 
 fn main() {
-    App::new()
-        .add_plugins(DefaultPlugins)
-        .add_systems(Startup, setup)
-        .run();
+    let mut app = App::new();
+
+    app.add_plugins(DefaultPlugins);
+
+    // Fallible systems can be used the same way as regular systems. The only difference is they
+    // return a `Result<(), Box<dyn Error>>` instead of a `()` (unit) type. Bevy will handle both
+    // types of systems the same way, except for the error handling.
+    app.add_systems(Startup, (setup, failing_system));
+
+    // By default, fallible systems that return an error will panic.
+    //
+    // We can change this by setting a custom error handler. This can be done globally for all
+    // systems in a given `App`. Here we set the global error handler using one of the built-in
+    // error handlers. Bevy provides built-in handlers for `panic`, `error`, `warn`, `info`,
+    // `debug`, `trace` and `ignore`.
+    app.set_systems_error_handler(bevy::ecs::result::warn);
```

Comment:
I slightly prefer `app.set_system_error_handler` and `DefaultSystemErrorHandler` over the plural form, as error handler selection happens from the perspective of a single system.

Reply:
Resolved in [7ac274a](https://github.com/bevyengine/bevy/pull/17753/commits/7ac274a3edda041444296e84f87e1bfbe6d783aa).